In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
import pandas as pd

In [ ]:
df = pd.read_csv('/content/drive/MyDrive/UFC/masterMLpublic.csv')

In [ ]:
import numpy as np
import xgboost as xgb
from sklearn.metrics import accuracy_score

Removing draw's, refplacing inftinte values with NaN and removing the NaN's

In [ ]:
#df = df[df['method'] != "DRAW"] # Remove draws
#df = df[df['division'] != "Women's Bantamweight"]
#df = df[df['division'] != "Heavyweight"]
#df = df[df['division'] != "Women's Flyweight"]
df = df.replace([np.inf, -np.inf], np.nan) # Replace inf with NaN
df = df.dropna() # Remove NaNs
df = df.loc[(df['date'] > '2021-11-01')]
#df = df[df.columns.drop(list(df.filter(like='odds')))] # Remove odds columns

In [ ]:
"""
test_df_svm = df.loc[(df['date'] > '2022-11-01')]
y_test_svm = test_df_svm['result']
X_test_svm = test_df_svm[['age', 'height','reach']]  # This will select the 'age' column

train_df_svm = df.drop(test_df_svm.index)
y_train_svm = train_df_svm['result']
X_train_age = train_df_svm[['age', 'height','reach']]  # Same as a
"""

"\ntest_df_svm = df.loc[(df['date'] > '2022-11-01')]\ny_test_svm = test_df_svm['result']\nX_test_svm = test_df_svm[['age', 'height','reach']]  # This will select the 'age' column\n\ntrain_df_svm = df.drop(test_df_svm.index)\ny_train_svm = train_df_svm['result'] \nX_train_age = train_df_svm[['age', 'height','reach']]  # Same as a\n"

In [ ]:
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score, classification_report
from sklearn.preprocessing import StandardScaler

In [ ]:
y = df['result']
"""
X = df[['age', 'height','reach','precomp_age_vs_opp', 'precomp_avg_head_strikes_absorbed_differential_vs_opp', 'precomp_elo_differential', 'precomp_age_differential_vs_opp', 'precomp_elo_vs_opp',
        'precomp_change_avg_elo_differential_vs_valley', 'precomp_recent_avg_age_vs_opp', 'precomp_recent_avg_head_strikes_absorbed_differential_valley_vs_opp', 'precomp_age_differential',
        'precomp_avg_age_vs_opp', 'precomp_change_avg_elo_differential', 'precomp_recent_avg_ground_strikes_attempts_differential_valley_vs_opp', 'precomp_avg_sig_strikes_absorbed_differential_vs_opp',
        'precomp_recent_avg_age_differential_vs_opp','precomp_change_avg_age_differential_vs_opp', 'precomp_head_strikes_landed_differential_valley_vs_opp', 'precomp_change_recent_avg_elo_differential_vs_peak',
        'precomp_change_avg_elo_differential_vs_peak', 'precomp_elo_differential_vs_peak', 'precomp_sig_strikes_landed_differential_valley_vs_opp', 'precomp_avg_ground_strikes_absorbed_differential_vs_opp',
        'precomp_avg_takedowns_landed_differential_peak_vs_opp', 'precomp_change_age_differential_vs_opp', 'precomp_recent_avg_age_differential', 'precomp_avg_control_differential_vs_opp', 'precomp_avg_age_differential_vs_opp',
        'precomp_recent_avg_sig_strikes_absorbed_differential_peak_vs_opp', 'precomp_recent_avg_sig_strikes_absorbed_differential_vs_opp', 'precomp_recent_avg_control_valley_vs_opp', 'precomp_change_elo_differential_vs_peak',
        'precomp_avg_distance_strikes_absorbed_differential_vs_opp', 'precomp_recent_avg_ground_strikes_landed_differential_valley_vs_opp', 'precomp_change_elo_differential_vs_valley', 'precomp_head_strikes_def_vs_opp',
        'precomp_elo_valley_vs_opp', 'precomp_head_strikes_def_differential_vs_opp', 'precomp_recent_avg_sig_strikes_absorbed_differential_valley_vs_opp' ,'precomp_win_loss_ratio_differential',
        'precomp_avg_head_strikes_landed_per_min_differential_vs_opp', 'precomp_recent_avg_head_strikes_absorbed_differential_peak_vs_opp' ,'precomp_avg_sig_strikes_absorbed_peak_vs_opp', 'precomp_sig_strikes_absorbed_differential_peak_vs_opp' ,
        'precomp_trueskill_vs_opp', 'precomp_change_recent_avg_elo_differential', 'precomp_avg_head_strikes_landed_differential_vs_opp', 'precomp_takedowns_attempts_per_min_peak_vs_opp', 'precomp_change_recent_avg_ground_strikes_attempts_per_min_differential_vs_valley_vs_opp',
        'precomp_change_recent_avg_ground_strikes_attempts_per_min_differential_vs_peak_vs_opp', 'precomp_change_recent_avg_ground_strikes_attempts_per_min_differential_valley_vs_opp', 'precomp_change_recent_avg_ground_strikes_attempts_per_min_differential_peak_vs_opp',
        'precomp_change_avg_ground_strikes_attempts_per_min_differential_vs_valley_vs_opp', 'precomp_change_avg_ground_strikes_attempts_per_min_differential_vs_peak_vs_opp', 'precomp_change_avg_ground_strikes_attempts_per_min_differential_valley_vs_opp',
        'precomp_change_avg_ground_strikes_attempts_per_min_differential_peak_vs_opp', 'precomp_change_recent_avg_ground_strikes_landed_per_min_differential_vs_valley_vs_opp', 'precomp_change_recent_avg_ground_strikes_landed_per_min_differential_vs_peak_vs_opp',
        'precomp_change_recent_avg_ground_strikes_landed_per_min_differential_valley_vs_opp', 'precomp_change_recent_avg_ground_strikes_landed_per_min_differential_peak_vs_opp', 'precomp_change_avg_ground_strikes_landed_per_min_differential_vs_valley_vs_opp',
        'precomp_change_avg_ground_strikes_landed_per_min_differential_vs_peak_vs_opp', 'precomp_change_avg_ground_strikes_landed_per_min_differential_valley_vs_opp', 'precomp_change_avg_ground_strikes_landed_per_min_differential_peak_vs_opp',
        'precomp_change_recent_avg_clinch_strikes_attempts_per_min_differential_vs_valley_vs_opp', 'precomp_change_recent_avg_clinch_strikes_attempts_per_min_differential_vs_peak_vs_opp', 'precomp_change_recent_avg_clinch_strikes_attempts_per_min_differential_valley_vs_opp',
        'precomp_change_recent_avg_clinch_strikes_attempts_per_min_differential_peak_vs_opp', 'precomp_change_avg_clinch_strikes_attempts_per_min_differential_vs_valley_vs_opp', 'precomp_change_avg_clinch_strikes_attempts_per_min_differential_vs_peak_vs_opp',
        'precomp_change_avg_clinch_strikes_attempts_per_min_differential_valley_vs_opp', 'precomp_change_avg_clinch_strikes_attempts_per_min_differential_peak_vs_opp', 'precomp_change_recent_avg_clinch_strikes_landed_per_min_differential_vs_valley_vs_opp',
        'precomp_change_recent_avg_clinch_strikes_landed_per_min_differential_vs_peak_vs_opp']]
"""
X = df[['age', 'height','reach','precomp_age_vs_opp', 'precomp_avg_head_strikes_absorbed_differential_vs_opp', 'precomp_elo_differential', 'precomp_age_differential_vs_opp', 'precomp_elo_vs_opp',
        'precomp_change_avg_elo_differential_vs_valley', 'precomp_recent_avg_age_vs_opp', 'precomp_recent_avg_head_strikes_absorbed_differential_valley_vs_opp', 'precomp_age_differential',
        'precomp_avg_age_vs_opp', 'precomp_change_avg_elo_differential', 'precomp_recent_avg_ground_strikes_attempts_differential_valley_vs_opp', 'precomp_avg_sig_strikes_absorbed_differential_vs_opp',
        'precomp_recent_avg_age_differential_vs_opp','precomp_change_avg_age_differential_vs_opp', 'precomp_head_strikes_landed_differential_valley_vs_opp', 'precomp_change_recent_avg_elo_differential_vs_peak',
        'precomp_change_avg_elo_differential_vs_peak', 'precomp_elo_differential_vs_peak', 'precomp_sig_strikes_landed_differential_valley_vs_opp', 'precomp_avg_ground_strikes_absorbed_differential_vs_opp',
        'precomp_avg_takedowns_landed_differential_peak_vs_opp', 'precomp_change_age_differential_vs_opp', 'precomp_recent_avg_age_differential', 'precomp_avg_control_differential_vs_opp', 'precomp_avg_age_differential_vs_opp',
        'precomp_recent_avg_sig_strikes_absorbed_differential_peak_vs_opp', 'precomp_recent_avg_sig_strikes_absorbed_differential_vs_opp', 'precomp_recent_avg_control_valley_vs_opp', 'precomp_change_elo_differential_vs_peak',
        'precomp_avg_distance_strikes_absorbed_differential_vs_opp', 'precomp_recent_avg_ground_strikes_landed_differential_valley_vs_opp', 'precomp_change_elo_differential_vs_valley', 'precomp_head_strikes_def_vs_opp',
        'precomp_elo_valley_vs_opp', 'precomp_head_strikes_def_differential_vs_opp', 'precomp_recent_avg_sig_strikes_absorbed_differential_valley_vs_opp' ,'precomp_win_loss_ratio_differential',
        'precomp_avg_head_strikes_landed_per_min_differential_vs_opp', 'precomp_recent_avg_head_strikes_absorbed_differential_peak_vs_opp' ,'precomp_avg_sig_strikes_absorbed_peak_vs_opp', 'precomp_sig_strikes_absorbed_differential_peak_vs_opp' ,
        'precomp_trueskill_vs_opp', 'precomp_change_recent_avg_elo_differential', 'precomp_avg_head_strikes_landed_differential_vs_opp', 'precomp_takedowns_attempts_per_min_peak_vs_opp', 'precomp_change_recent_avg_ground_strikes_attempts_per_min_differential_vs_valley_vs_opp',
        'precomp_change_recent_avg_ground_strikes_attempts_per_min_differential_vs_peak_vs_opp', 'precomp_change_recent_avg_ground_strikes_attempts_per_min_differential_valley_vs_opp', 'precomp_change_recent_avg_ground_strikes_attempts_per_min_differential_peak_vs_opp',
        'precomp_change_avg_ground_strikes_attempts_per_min_differential_vs_valley_vs_opp', 'precomp_change_avg_ground_strikes_attempts_per_min_differential_vs_peak_vs_opp', 'precomp_change_avg_ground_strikes_attempts_per_min_differential_valley_vs_opp',
        'precomp_change_avg_ground_strikes_attempts_per_min_differential_peak_vs_opp', 'precomp_change_recent_avg_ground_strikes_landed_per_min_differential_vs_valley_vs_opp',
        'precomp_change_recent_avg_ground_strikes_landed_per_min_differential_peak_vs_opp', 'precomp_change_avg_ground_strikes_landed_per_min_differential_vs_valley_vs_opp',
        'precomp_change_avg_ground_strikes_landed_per_min_differential_valley_vs_opp', 'precomp_change_avg_ground_strikes_landed_per_min_differential_peak_vs_opp',
        'precomp_change_recent_avg_clinch_strikes_attempts_per_min_differential_vs_valley_vs_opp', 'precomp_change_recent_avg_clinch_strikes_attempts_per_min_differential_vs_peak_vs_opp', 'precomp_change_recent_avg_clinch_strikes_attempts_per_min_differential_valley_vs_opp',
        'precomp_change_avg_clinch_strikes_attempts_per_min_differential_vs_valley_vs_opp', 'precomp_change_avg_clinch_strikes_attempts_per_min_differential_vs_peak_vs_opp',
        'precomp_change_avg_clinch_strikes_attempts_per_min_differential_valley_vs_opp', 'precomp_change_avg_clinch_strikes_attempts_per_min_differential_peak_vs_opp',
        'precomp_change_recent_avg_clinch_strikes_landed_per_min_differential_vs_peak_vs_opp']]

print("X shape: ", X.shape)
print("Y shape: ", y.shape)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 21)

X shape:  (636, 70)
Y shape:  (636,)


In [ ]:
from sklearn.decomposition import PCA
from sklearn.preprocessing import StandardScaler
from sklearn.svm import SVC
from sklearn.pipeline import make_pipeline
from sklearn.model_selection import train_test_split

In [ ]:
pipeline = make_pipeline(StandardScaler(), PCA(n_components=0.95), SVC(kernel='rbf'))

In [ ]:
pipeline.fit(X_train, y_train)

Pipeline(steps=[('standardscaler', StandardScaler()),
                ('pca', PCA(n_components=0.95)), ('svc', SVC())])

In [ ]:
accuracy = pipeline.score(X_test, y_test)
print(f'Accuracy: {accuracy}')

Accuracy: 0.6015625


In [395]:
from sklearn.decomposition import PCA
from sklearn.preprocessing import StandardScaler
from sklearn.svm import SVC
from sklearn.pipeline import make_pipeline
from sklearn.model_selection import train_test_split

# Define your base features that you will always include
base_features = ['age', 'height', 'reach', 'precomp_change_avg_elo_differential', 'precomp_avg_age_differential_vs_opp', 'precomp_change_avg_ground_strikes_attempts_per_min_differential_peak_vs_opp', 'precomp_change_avg_ground_strikes_landed_per_min_differential_peak_vs_opp', 'precomp_head_strikes_def_vs_opp', 'precomp_change_avg_clinch_strikes_attempts_per_min_differential_vs_peak_vs_opp', 'precomp_avg_control_differential_vs_opp', 'precomp_recent_avg_ground_strikes_landed_differential_valley_vs_opp'] # Add your actual base features

# List additional features to iterate through
additional_features = ['precomp_age_vs_opp', 'precomp_avg_head_strikes_absorbed_differential_vs_opp', 'precomp_elo_differential', 'precomp_age_differential_vs_opp', 'precomp_elo_vs_opp',
        'precomp_change_avg_elo_differential_vs_valley', 'precomp_recent_avg_age_vs_opp', 'precomp_recent_avg_head_strikes_absorbed_differential_valley_vs_opp', 'precomp_age_differential',
        'precomp_avg_age_vs_opp', 'precomp_change_avg_elo_differential', 'precomp_recent_avg_ground_strikes_attempts_differential_valley_vs_opp', 'precomp_avg_sig_strikes_absorbed_differential_vs_opp',
        'precomp_recent_avg_age_differential_vs_opp','precomp_change_avg_age_differential_vs_opp', 'precomp_head_strikes_landed_differential_valley_vs_opp', 'precomp_change_recent_avg_elo_differential_vs_peak',
        'precomp_change_avg_elo_differential_vs_peak', 'precomp_elo_differential_vs_peak', 'precomp_sig_strikes_landed_differential_valley_vs_opp', 'precomp_avg_ground_strikes_absorbed_differential_vs_opp',
        'precomp_avg_takedowns_landed_differential_peak_vs_opp', 'precomp_change_age_differential_vs_opp', 'precomp_recent_avg_age_differential', 'precomp_avg_control_differential_vs_opp', 'precomp_avg_age_differential_vs_opp',
        'precomp_recent_avg_sig_strikes_absorbed_differential_peak_vs_opp', 'precomp_recent_avg_sig_strikes_absorbed_differential_vs_opp', 'precomp_recent_avg_control_valley_vs_opp', 'precomp_change_elo_differential_vs_peak',
        'precomp_avg_distance_strikes_absorbed_differential_vs_opp', 'precomp_recent_avg_ground_strikes_landed_differential_valley_vs_opp', 'precomp_change_elo_differential_vs_valley', 'precomp_head_strikes_def_vs_opp',
        'precomp_elo_valley_vs_opp', 'precomp_head_strikes_def_differential_vs_opp', 'precomp_recent_avg_sig_strikes_absorbed_differential_valley_vs_opp' ,'precomp_win_loss_ratio_differential',
        'precomp_avg_head_strikes_landed_per_min_differential_vs_opp', 'precomp_recent_avg_head_strikes_absorbed_differential_peak_vs_opp' ,'precomp_avg_sig_strikes_absorbed_peak_vs_opp', 'precomp_sig_strikes_absorbed_differential_peak_vs_opp' ,
        'precomp_trueskill_vs_opp', 'precomp_change_recent_avg_elo_differential', 'precomp_avg_head_strikes_landed_differential_vs_opp', 'precomp_takedowns_attempts_per_min_peak_vs_opp', 'precomp_change_recent_avg_ground_strikes_attempts_per_min_differential_vs_valley_vs_opp',
        'precomp_change_recent_avg_ground_strikes_attempts_per_min_differential_vs_peak_vs_opp', 'precomp_change_recent_avg_ground_strikes_attempts_per_min_differential_valley_vs_opp', 'precomp_change_recent_avg_ground_strikes_attempts_per_min_differential_peak_vs_opp',
        'precomp_change_avg_ground_strikes_attempts_per_min_differential_vs_valley_vs_opp', 'precomp_change_avg_ground_strikes_attempts_per_min_differential_vs_peak_vs_opp', 'precomp_change_avg_ground_strikes_attempts_per_min_differential_valley_vs_opp',
        'precomp_change_avg_ground_strikes_attempts_per_min_differential_peak_vs_opp', 'precomp_change_recent_avg_ground_strikes_landed_per_min_differential_vs_valley_vs_opp', 'precomp_change_recent_avg_ground_strikes_landed_per_min_differential_vs_peak_vs_opp',
        'precomp_change_recent_avg_ground_strikes_landed_per_min_differential_valley_vs_opp', 'precomp_change_recent_avg_ground_strikes_landed_per_min_differential_peak_vs_opp', 'precomp_change_avg_ground_strikes_landed_per_min_differential_vs_valley_vs_opp',
        'precomp_change_avg_ground_strikes_landed_per_min_differential_vs_peak_vs_opp', 'precomp_change_avg_ground_strikes_landed_per_min_differential_valley_vs_opp', 'precomp_change_avg_ground_strikes_landed_per_min_differential_peak_vs_opp',
        'precomp_change_recent_avg_clinch_strikes_attempts_per_min_differential_vs_valley_vs_opp', 'precomp_change_recent_avg_clinch_strikes_attempts_per_min_differential_vs_peak_vs_opp', 'precomp_change_recent_avg_clinch_strikes_attempts_per_min_differential_valley_vs_opp',
        'precomp_change_recent_avg_clinch_strikes_attempts_per_min_differential_peak_vs_opp', 'precomp_change_avg_clinch_strikes_attempts_per_min_differential_vs_valley_vs_opp', 'precomp_change_avg_clinch_strikes_attempts_per_min_differential_vs_peak_vs_opp',
        'precomp_change_avg_clinch_strikes_attempts_per_min_differential_valley_vs_opp', 'precomp_change_avg_clinch_strikes_attempts_per_min_differential_peak_vs_opp', 'precomp_change_recent_avg_clinch_strikes_landed_per_min_differential_vs_valley_vs_opp',
        'precomp_change_recent_avg_clinch_strikes_landed_per_min_differential_vs_peak_vs_opp'] # Add your actual additional features

# Initialize the best accuracy to a very low number
best_accuracy = 0
best_feature_set = []

# Iterate through additional features
for feature in additional_features:
    # Create a new feature set including the new feature
    current_features = base_features + [feature]

    # Split the data into training and testing sets
    X_train, X_test, y_train, y_test = train_test_split(
        df[current_features], y, test_size=0.2, random_state=42
    )

    # Create a pipeline that standardizes, runs PCA, and trains SVM
    pipeline = make_pipeline(StandardScaler(), PCA(n_components=0.95), SVC(kernel='rbf'))

    # Fit the model
    pipeline.fit(X_train, y_train)

    # Calculate the accuracy
    current_accuracy = pipeline.score(X_test, y_test)

    # Check if the current feature set gave a better accuracy
    if current_accuracy > best_accuracy:
        best_accuracy = current_accuracy
        best_feature_set = current_features

# Print the best accuracy and feature set
print(f'Best Accuracy: {best_accuracy}')
print(f'Best Feature Set: {best_feature_set}')

Best Accuracy: 0.78125
Best Feature Set: ['age', 'height', 'reach', 'precomp_change_avg_elo_differential', 'precomp_avg_age_differential_vs_opp', 'precomp_change_avg_ground_strikes_attempts_per_min_differential_peak_vs_opp', 'precomp_change_avg_ground_strikes_landed_per_min_differential_peak_vs_opp', 'precomp_head_strikes_def_vs_opp', 'precomp_change_avg_clinch_strikes_attempts_per_min_differential_vs_peak_vs_opp', 'precomp_avg_control_differential_vs_opp', 'precomp_recent_avg_ground_strikes_landed_differential_valley_vs_opp', 'precomp_avg_head_strikes_landed_per_min_differential_vs_opp']


In [396]:
from itertools import combinations
from sklearn.decomposition import PCA
from sklearn.preprocessing import StandardScaler
from sklearn.svm import SVC
from sklearn.pipeline import make_pipeline
from sklearn.model_selection import train_test_split

# Define your base features that you will always include
base_features = ['age', 'height','reach']  # Add your actual base features

# List additional features to iterate through
additional_features = ['precomp_change_avg_elo_differential', 'precomp_avg_age_differential_vs_opp', 'precomp_change_avg_ground_strikes_attempts_per_min_differential_peak_vs_opp', 'precomp_change_avg_ground_strikes_landed_per_min_differential_peak_vs_opp', 'precomp_head_strikes_def_vs_opp', 'precomp_change_avg_clinch_strikes_attempts_per_min_differential_vs_peak_vs_opp', 'precomp_avg_control_differential_vs_opp', 'precomp_recent_avg_ground_strikes_landed_differential_valley_vs_opp', 'precomp_avg_head_strikes_landed_per_min_differential_vs_opp']  # Add your actual additional features
#,'precomp_age_vs_opp','precomp_head_strikes_def_vs_opp','precomp_age_vs_opp','precomp_change_avg_clinch_strikes_attempts_per_min_differential_vs_peak_vs_opp','precomp_avg_control_differential_vs_opp','precomp_change_recent_avg_ground_strikes_attempts_per_min_differential_peak_vs_opp','precomp_recent_avg_ground_strikes_landed_differential_valley_vs_opp'
# Initialize the best accuracy to a very low number
best_accuracy = 0
best_feature_set = []

# Generate all possible combinations of additional features
for i in range(1, len(additional_features) + 1):
    for combo in combinations(additional_features, i):
        # Create a new feature set including the new combination of features
        current_features = base_features + list(combo)

        # Split the data into training and testing sets
        X_train, X_test, y_train, y_test = train_test_split(
            df[current_features], y, test_size=0.2, random_state=42
        )

        # Create a pipeline that standardizes, runs PCA, and trains SVM
        pipeline = make_pipeline(StandardScaler(), PCA(n_components=0.95), SVC(kernel='rbf'))

        # Fit the model
        pipeline.fit(X_train, y_train)

        # Calculate the accuracy
        current_accuracy = pipeline.score(X_test, y_test)

        # Check if the current combination of features gave a better accuracy
        if current_accuracy > best_accuracy:
            best_accuracy = current_accuracy
            best_feature_set = current_features

# Print the best accuracy and feature set
print(f'Best Accuracy: {best_accuracy}')
print(f'Best Feature Set: {best_feature_set}')


Best Accuracy: 0.78125
Best Feature Set: ['age', 'height', 'reach', 'precomp_change_avg_elo_differential', 'precomp_avg_age_differential_vs_opp', 'precomp_change_avg_ground_strikes_attempts_per_min_differential_peak_vs_opp', 'precomp_change_avg_ground_strikes_landed_per_min_differential_peak_vs_opp', 'precomp_head_strikes_def_vs_opp', 'precomp_change_avg_clinch_strikes_attempts_per_min_differential_vs_peak_vs_opp', 'precomp_avg_control_differential_vs_opp', 'precomp_recent_avg_ground_strikes_landed_differential_valley_vs_opp']


Using only age, reach, and height with a pipeline, standardscaler, PCA with 0.95 components, and rbf kernel got an accuracy os 0.47656

(+)with precomp_age_vs_opp: 0.5

(+) with precomp_avg_head_strikes_absorbed_differential_vs_opp: 0.5234375

(=)with precomp_elo_differential: 0.5625

(+)with precomp_age_differential_vs_opp: 0.5625

(-)with precomp_elo_vs_opp: 0.5546875

(+)with precomp_change_avg_elo_differential_vs_valley: 0.578125

(=)with precomp_recent_avg_age_vs_opp: 0.578125

(=)with precomp_recent_avg_head_strikes_absorbed_differential_valley_vs_opp: 0.578125

(-)with precomp_age_differential: 0.5703125

(+) with precomp_avg_age_vs_opp: 0.578125

(=) with precomp_change_avg_elo_differential: 0.578125

(-) with precomp_recent_avg_ground_strikes_attempts_differential_valley_vs_opp: 0.5546875

(=) with precomp_avg_sig_strikes_absorbed_differential_vs_opp: 0.5546875

(+) with precomp_recent_avg_age_differential_vs_opp: 0.5625

(+) with precomp_change_avg_age_differential_vs_opp: 0.5859375

(-) with precomp_head_strikes_landed_differential_valley_vs_opp: 0.578125

(-) with precomp_change_recent_avg_elo_differential_vs_peak: 0.5703125

(+) with precomp_change_avg_elo_differential_vs_peak: 0.609375

(=) with precomp_elo_differential_vs_peak: 0.609375

(+) with precomp_sig_strikes_landed_differential_valley_vs_opp: 0.625

(-) with precomp_avg_ground_strikes_absorbed_differential_vs_opp: 0.625

(-) with precomp_avg_takedowns_landed_differential_peak_vs_opp: 0.609375

(-) with precomp_change_age_differential_vs_opp: 0.578125

(-) with precomp_recent_avg_age_differential: 0.5703125

(+) with precomp_avg_control_differential_vs_opp: 0.59375

(=) with precomp_avg_age_differential_vs_opp: 0.59375

(-) with precomp_recent_avg_sig_strikes_absorbed_differential_peak_vs_opp: 0.5859375

(+) with precomp_recent_avg_sig_strikes_absorbed_differential_vs_opp: 0.59375

(+) with precomp_recent_avg_control_valley_vs_opp: 0.609375

(=) with precomp_change_elo_differential_vs_peak: 0.609375

(-) with precomp_avg_distance_strikes_absorbed_differential_vs_opp: 0.59375

(+) with precomp_recent_avg_ground_strikes_landed_differential_valley_vs_opp: 0.640625

*(+) with precomp_change_elo_differential_vs_valley: 0.6484375

(-) with precomp_head_strikes_def_vs_opp: 0.640625

(-) with precomp_elo_valley_vs_opp: 0.609375

(+) with precomp_head_strikes_def_differential_vs_opp: 0.625

(=) with precomp_recent_avg_sig_strikes_absorbed_differential_valley_vs_opp: 0.625

(-) with precomp_win_loss_ratio_differential: 0.6171875

(-) with precomp_avg_head_strikes_landed_per_min_differential_vs_opp: 0.6015625

(+) with precomp_recent_avg_head_strikes_absorbed_differential_peak_vs_opp: 0.609375

(-) with precomp_avg_sig_strikes_absorbed_peak_vs_opp: 0.59375

(=) with precomp_sig_strikes_absorbed_differential_peak_vs_opp: 0.59375

(=) with precomp_trueskill_vs_opp: 0.59375

(+) with precomp_change_recent_avg_elo_differential: 0.609375

(-) with precomp_avg_head_strikes_landed_differential_vs_opp: 0.5859375

(+) with precomp_takedowns_attempts_per_min_peak_vs_opp: 0.59375

(+) with precomp_change_recent_avg_ground_strikes_attempts_per_min_differential_vs_valley_vs_opp: 0.6015625

(-) with precomp_change_recent_avg_ground_strikes_attempts_per_min_differential_vs_peak_vs_opp: 0.5859375

(-) with precomp_change_recent_avg_ground_strikes_attempts_per_min_differential_valley_vs_opp: 0.5703125

(+) with precomp_change_recent_avg_ground_strikes_attempts_per_min_differential_peak_vs_opp: 0.59375

(=) with precomp_change_avg_ground_strikes_attempts_per_min_differential_vs_valley_vs_opp: 0.59375

(+) with precomp_change_avg_ground_strikes_attempts_per_min_differential_vs_peak_vs_opp: 0.6015625

(=) with precomp_change_avg_ground_strikes_attempts_per_min_differential_valley_vs_opp: 0.6015625

(-) with precomp_change_avg_ground_strikes_attempts_per_min_differential_peak_vs_opp: 0.578125

(+) with precomp_change_recent_avg_ground_strikes_landed_per_min_differential_vs_valley_vs_opp: 0.6015625

(-) with precomp_change_recent_avg_ground_strikes_landed_per_min_differential_vs_peak_vs_opp: 0.5859375

(-) with precomp_change_recent_avg_ground_strikes_landed_per_min_differential_valley_vs_opp: 0.578125

(+) with precomp_change_recent_avg_ground_strikes_landed_per_min_differential_peak_vs_opp: 0.5859375

(+) with precomp_change_avg_ground_strikes_landed_per_min_differential_vs_valley_vs_opp: 0.59375

(-) with precomp_change_avg_ground_strikes_landed_per_min_differential_vs_peak_vs_opp: 0.5703125

(=) with precomp_change_avg_ground_strikes_landed_per_min_differential_valley_vs_opp: 0.5703125

(+) with precomp_change_avg_ground_strikes_landed_per_min_differential_peak_vs_opp: 0.578125

(+) with precomp_change_recent_avg_clinch_strikes_attempts_per_min_differential_vs_valley_vs_opp: 0.5859375

(+) with precomp_change_recent_avg_clinch_strikes_attempts_per_min_differential_vs_peak_vs_opp: 0.6171875

(+) with precomp_change_recent_avg_clinch_strikes_attempts_per_min_differential_valley_vs_opp: 0.625

(-) with precomp_change_recent_avg_clinch_strikes_attempts_per_min_differential_peak_vs_opp: 0.6015625

(+) with precomp_change_avg_clinch_strikes_attempts_per_min_differential_vs_valley_vs_opp: 0.609375

(+) with precomp_change_avg_clinch_strikes_attempts_per_min_differential_vs_peak_vs_opp: 0.625

(+) with precomp_change_avg_clinch_strikes_attempts_per_min_differential_valley_vs_opp: 0.6171875

(+) with precomp_change_avg_clinch_strikes_attempts_per_min_differential_peak_vs_opp: 0.6328125

(-) with precomp_change_recent_avg_clinch_strikes_landed_per_min_differential_vs_valley_vs_opp: 0.625

(=) with precomp_change_recent_avg_clinch_strikes_landed_per_min_differential_vs_peak_vs_opp: 0.625

(-) Removed precomp_change_recent_avg_clinch_strikes_landed_per_min_differential_vs_valley_vs_opp: 0.6171875

(=)Removed precomp_change_recent_avg_clinch_strikes_attempts_per_min_differential_peak_vs_opp: 0.6171875

(-) Removed precomp_change_avg_ground_strikes_landed_per_min_differential_vs_peak_vs_opp: 0.6015625

(+) Removed precomp_change_recent_avg_ground_strikes_landed_per_min_differential_valley_vs_opp: 0.609375

(-) Removed precomp_change_recent_avg_ground_strikes_landed_per_min_differential_vs_peak_vs_opp: 0.6015625